go-controller/pkg/ovn/policy.go

```diff
@@ -95,17 +98,22 @@ func (oc *Controller) syncNetworkPolicies(networkPolicies []interface{}) {
 	}
 }
 
-func addAllowACLFromNode(logicalSwitch string, mgmtPortIP net.IP) error {
+func addAllowACLFromNode(logicalSwitch string, mgmtPortIP net.IP, ovnNBClient goovn.Client) error {
```

Comment:
@abhat @trozet Please weigh in with your comments as well, Thanks

Reply:
iirc when @abhat first introduced using the go-ovn api we did have the clients as singletons, but then we decided that wasn't a great idea. I don't see a way other than passing it around to the functions that need it. @danwinship is probably the best person to bring into this conversation since he has a lot of expertise in golang code structure. What do you think Dan?

Reply:
> iirc when @abhat first introduced using the go-ovn api we did have the clients as singletons, but then we decided that wasn't a great idea.

Hm... I don't remember that.

`pkg/config` and `pkg/util/ovs.go` use singletons (which can be overridden for unit tests) and that seems to work ok?

> could cause difficulties if the same code needs to manipulate multiple NB and SB databases.

but I don't think we'd ever need to do that.


From all the code I've seen in kubernetes, I don't think anyone has figured out an optimal way to do this. All the solutions are ugly in their own way.

Reply:
The comment was I think from @dcbw. The passing of go-client as arguments is valuable because in future if we decide to slice and dice the ovn-controller (across namespaces, albeit hypothetically, for instance), you could like @anfredette pointed out create multiple nbdb/sbdb client instances one for each controller shard.

So while this is inconvenient and one may argue it is premature optimization at this point, we should try to keep that abstraction.